pkgs/os-specific/linux/raspberrypi-eeprom/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenvNoCC, lib, fetchFromGitHub, makeWrapper
+, python3, binutils-unwrapped, findutils, kmod, pciutils, raspberrypi-tools
+}:
+stdenvNoCC.mkDerivation {
+  pname = "raspberrypi-eeprom";
+  version = "unstable-2020-05-15"; # USB boot on the beta channel
+
+  src = fetchFromGitHub {
+    owner = "raspberrypi";
+    repo = "rpi-eeprom";
+    rev = "ad18a5b468f787ed37ab62e0a699dabeaa580e27";
+    sha256 = "sha256-E+Fq7VO+gLbCVAawuYfYxBFGER6nelgR8EHwSnc37GQ=";
+  };
+
+  buildInputs = [ python3 ];
+  nativeBuildInputs = [ makeWrapper ];
+
+  postPatch = ''
+    # Don't try to verify md5 signatures from /var/lib/dpkg and
+    # fix path to the configuration.
+    substituteInPlace rpi-eeprom-update \
+      --replace 'IGNORE_DPKG_CHECKSUMS=$LOCAL_MODE' 'IGNORE_DPKG_CHECKSUMS=1' \
+      --replace '/etc/default' '/etc'
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin $out/share/rpi-eeprom
+
+    cp rpi-eeprom-config rpi-eeprom-update $out/bin
+    cp -r firmware/{beta,critical,old,stable} $out/share/rpi-eeprom
+    cp -r firmware/vl805 $out/bin
+  '';
+
+  fixupPhase = ''
+    patchShebangs $out/bin
+    wrapProgram $out/bin/rpi-eeprom-update \
+      --set FIRMWARE_ROOT $out/share/rpi-eeprom \
+      --set VCMAILBOX ${raspberrypi-tools}/bin/vcmailbox \
+      --prefix PATH : "${lib.makeBinPath [ binutils-unwrapped findutils kmod pciutils raspberrypi-tools (placeholder "out") ]}"
```

Comment:
it may help to omit `raspberrypi-tools`,  `vl805` and maybe `rpi-eeprom-update` when targeting x86, since the `rpi-eeprom-config` script still works on x86, and is useful to have there

Reply:
How is `rpi-eeprom-config` useful on x86? What's the use-case?